### PR TITLE
[sdk] screen orientation js

### PR DESCRIPTION
**Pull request by quinlanj**
# Why

ScreenOrientation JS portion

Related PRS:
Docs - https://github.com/expo/universe/pull/3280
JS - https://github.com/expo/expo/pull/3103
iOS - https://github.com/expo/expo/pull/3104
Android - https://github.com/expo/expo/pull/3105
Tests - https://github.com/expo/expo/pull/3106
Previous Feedback (1st pass android) - https://github.com/expo/expo/pull/2832

### CODE REVIEW

**Review by ide — CHANGES_REQUESTED** [review]
I would really like to see Jest tests for this module, especially where there is meaningful logic or tricky edge cases.

**Review by esamelson — CHANGES_REQUESTED** [review]
agree with @ide's comments and adding a few more, but most of this looks great 🙏

**Review by ide — CHANGES_REQUESTED** [review]

File: packages/expo/src/ScreenOrientation/ScreenOrientation.ts [hunk 1]
```diff
@@ -1,30 +1,210 @@
...
+
+enum iOSSizeClass {
+  VERTICAL = 'VERTICAL',
+  HORIZONTAL = 'HORIZONTAL',
+  UNKNOWN = 'UNKNOWN',
+};
+
+type OrientationInfo = {
+  orientation: Orientation;
+  verticalSizeClass?: iOSSizeClass;
```
[the remark this answers is not in the document]
Comment (quinlanj) [reply packages/expo/src/ScreenOrientation/ScreenOrientation.ts]: you are correct -- good catch! the iOSSizeClasses are meant to be REGULAR/COMPACT, and each iOS device has a sizeClass for the vertical and horizontal axis
